uobtheatre/bookings/schema.py

```diff
@@ -634,34 +634,43 @@ def resolve_mutation(  # pylint: disable=too-many-arguments
         if booking.status != Booking.BookingStatus.IN_PROGRESS:
             raise GQLException(message="The booking is not in progress")
 
-        if payment_provider == SquareOnline.name:
-            if not nonce:
-                raise GQLException(
-                    message=f"A nonce is required when using {payment_provider} provider.",
-                    field="nonce",
-                    code="missing_required",
-                )
-            payment_method = SquareOnline(nonce, booking.id)
-
-        elif payment_provider == SquarePOS.name:
-            if not device_id:
-                raise GQLException(
-                    message=f"A device_id is required when using {payment_provider} provider.",
-                    field="device_id",
-                    code="missing_required",
+        # Booking must have at least one ticket
+        if booking.tickets.count() == 0:
+            raise GQLException(message="The booking must have at least one ticket")
+
+        # If the booking isn't free, we care about the payment provider. Otherwise, we don't
+        if booking.total() > 0:
```

Comment:
Can we not nest this? So if just the first branch in the `elif`?
```suggestion
        if booking.total() == 0:
            # Free bookings do not require a payment
            pass
```